lib/actions/appleMusic.js

```diff
@@ -155,15 +145,16 @@ function appleMusic(player, values) {
                 return player.coordinator.clearQueue()
                   .then(() => player.coordinator.addURIToQueue(queueTracks[0].uri, queueTracks[0].metadata, true, 1))
                   .then(() => {
-                      if (radioOn(player)) {
-                        return promise.then(() => player.coordinator.setAVTransport(queueURI,''));
+                      if (isRadioOrLineIn(player.coordinator.state.currentTrack.uri)) {
+                        player.coordinator.setAVTransport(queueURI,''));
                       }
+                      return Promise.resolve();
                     })
                   .then(() => player.coordinator.play())
                   .then(() => {
                     return queueTracks.reduce(function(promise, track, index){
                       if (index > 0) {
-                        return promise.then(() => player.coordinator.addURIToQueue(track.uri, track.metadata, true, index + 1));
+                        player.coordinator.addURIToQueue(track.uri, track.metadata, true, index + 1);
                       } else {
                         return promise.then(() => {});
                       }   
```

Comment:
Thanks Jimmy, I see what I did. I was trying to fix something else and deleted that one too by mistake.  

I'm still obviously not totally clear on Promises yet, so seeing some more examples will be helpful if you can indeed modify what I have for the desired behavior.  The desire is to do a quick return after playing the first song, and then add the rest of the tracks at an orderly request/wait-for-response pace.

A couple of Promises questions that I have:

1. When you have a single function call for a .then(() => aFunction());, you do not seem to need to do a return, but when the function is bracketed, you are saying that you do need to perform a return.  Is that correct?

2. When I tried to return promise.then(() => player.coordinator.XXX... in the cases that I had to go back and fix, or even if I returned the function directly, it was throwing a then/Promise does not exist error.  Yet in other areas of the code it seems like the functions can be returned without this exception.  Once again it seems like the {}'s are in play here.  Can you point me to some documentation on the whole (() => function OR {} syntax ?  I'm having trouble finding anything on this.